velox/vector/VectorPool.cpp

```diff
@@ -24,7 +24,7 @@ namespace {
 /// type is not supported, i.e. not a built-in singleton type.
 FOLLY_ALWAYS_INLINE int32_t toCacheIndex(const TypePtr& type) {
   static constexpr int32_t kNumCachedVectorTypes =
-      static_cast<int32_t>(TypeKind::INTERVAL_DAY_TIME) + 1;
+      static_cast<int32_t>(TypeKind::DATE) + 1;
```

Comment:
@mbasmanova : Why did you use TYPEKIND::DATE here ?

Reply:
@aditi-pandit After removing INTERVAL_DAY_TIME, DATE is the last supported TypeKind.